src/main/java/duke/functions/TaskList.java

```diff
@@ -0,0 +1,146 @@
+package duke.functions;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import duke.datatypes.Deadline;
+import duke.datatypes.Event;
+import duke.datatypes.Task;
+import duke.datatypes.Todo;
+
+public class TaskList {
+    private ArrayList<Task> arrayOfTasks = new ArrayList<>();
+    private final FileIO savedTextFile = new FileIO("data/savedtasklist.txt");
+    private int numberOfTasks;
+
+    public TaskList() {
+        setNumberOfTasks(0);
+    }
+
+    public int getNumberOfTasks() {
+        return numberOfTasks;
+    }
+
+    public void setNumberOfTasks(int numberOfTasks) {
+        this.numberOfTasks=numberOfTasks;
+    }
+
+    public void printAllTasks() {
+        if(arrayOfTasks.isEmpty()) {
+            System.out.println("OOPS!!! No task added yet.");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i<arrayOfTasks.size();i++){
```

Comment:
You may want to include spacing between "i" and "<" and ";" and "i" to abide by the coding standards